requestmanager/asyncloader/asyncloader.go

```diff
@@ -347,13 +364,16 @@ func (crm *cleanupRequestMessage) handle(al *AsyncLoader) {
 	al.responseCache.FinishRequest(crm.requestID)
 }
 
-func setupAttemptQueue(loader ipld.Loader, storer ipld.Storer) (*responsecache.ResponseCache, *loadattemptqueue.LoadAttemptQueue) {
+func setupAttemptQueue(loader ipld.Loader, storer ipld.Storer, allocator Allocator) (*responsecache.ResponseCache, *loadattemptqueue.LoadAttemptQueue) {
 
 	unverifiedBlockStore := unverifiedblockstore.New(storer)
 	responseCache := responsecache.New(unverifiedBlockStore)
-	loadAttemptQueue := loadattemptqueue.New(func(requestID graphsync.RequestID, link ipld.Link) types.AsyncLoadResult {
+	loadAttemptQueue := loadattemptqueue.New(func(p peer.ID, requestID graphsync.RequestID, link ipld.Link) types.AsyncLoadResult {
 		// load from response cache
 		data, err := responseCache.AttemptLoad(requestID, link)
+		if data != nil {
+			allocator.ReleaseBlockMemory(p, uint64(len(data)))
```

Comment:
I was wondering what errors could `ReleaseBlockMemory` return. The only place in current code where we catch the error seems to be in [`allocator_test`](https://github.com/ipfs/go-graphsync/blob/feat/memory_pressure_incoming_messages/allocator/allocator_test.go#L329). If not caught anywhere else why have `error` return type in signature at all?

Another numpty question: judging by the function names, i.e. `setup` and `load` I was expecting memory allocation instead of release; I am conscious I'm new to graphsync and learning so likely I'm missing something.

Reply:
re: `setup` and `load`

You're totally right it seems odd. Perhaps we should work on a different name. Concretely, this is the moment we actually verify that the block we've received from the other peer is part of the selector traversal we specified -- so:
- it's a load in the sense that it's being requested by the selector traversal we're using to verified the incoming data
- it's the moment the block received on the network moves to permaneant storage, meaning it's released from the temporary unverified blockstore. 
- agree the naming is confusing and probably deserves iteration

Reply:
BTW, I love iterating on names cause I never get it right the first time (hard problem in CS and all) so always feel free to suggest things :)

Reply:
> iterating 
>  (hard problem in CS and all) so always feel free to suggest things

totally!
pft... I don't stand a chance in coming up with better names; maybe later once I have a better grasp the flow :)